tests/components/home_connect/test_init.py

```diff
@@ -0,0 +1,208 @@
+    {
+        "domain": DOMAIN,
+        "service": "set_option_active",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+            "key": "",
+            "value": "",
+            "unit": "",
+        },
+        "blocking": True,
+    },
+    {
+        "domain": DOMAIN,
+        "service": "set_option_selected",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+            "key": "",
+            "value": "",
+        },
+        "blocking": True,
+    },
+    {
+        "domain": DOMAIN,
+        "service": "change_setting",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+            "key": "",
+            "value": "",
+        },
+        "blocking": True,
+    },
+]
+
+SERVICE_COMMAND_CALL_PARAMS = [
+    {
+        "domain": DOMAIN,
+        "service": "pause_program",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+        },
+        "blocking": True,
+    },
+    {
+        "domain": DOMAIN,
+        "service": "resume_program",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+        },
+        "blocking": True,
+    },
+]
+
+
+SERVICE_PROGRAM_CALL_PARAMS = [
+    {
+        "domain": DOMAIN,
+        "service": "select_program",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+            "program": "",
+            "key": "",
+            "value": "",
+        },
+        "blocking": True,
+    },
+    {
+        "domain": DOMAIN,
+        "service": "start_program",
+        "service_data": {
+            "device_id": "DEVICE_ID",
+            "program": "",
+            "key": "",
+            "value": "",
+            "unit": "C",
+        },
+        "blocking": True,
+    },
+]
+
+
+async def test_setup(
+    config_entry_auth,
+    hass: HomeAssistant,
+    integration_setup: Callable[[], Awaitable[bool]],
+    config_entry: MockConfigEntry,
+    setup_credentials: None,
+) -> None:
+    """Test setting up the integration."""
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
+
+    assert await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+
+
+@patch.object(ConfigEntryAuth, "get_devices")
+async def test_update_throttle(
+    get_devices,
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    platforms: list[Platform],
+) -> None:
+    """Test to check Throttle functionality."""
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+
+    assert await integration_setup()
+    assert get_devices.call_count == 0
+    assert config_entry.state == ConfigEntryState.LOADED
+
+
+@patch.object(
+    ConfigEntryAuth, "get_devices", side_effect=HTTPError(response=MagicMock())
```

Comment:
Don't patch `get_devices` which is our code. Patch the library, eg `get_appliances`.